WcaOnRails/app/webpacker/components/EditSchedule/EditActivities/index.js

```diff
@@ -0,0 +1,443 @@
+
+    const duration = toLuxonDuration(delta, calendar);
+    const deltaIso = duration.toISO();
+
+    dispatch(moveActivity(activityId, deltaIso));
+  };
+
+  const resizeActivity = ({
+    event: fcEvent,
+    startDelta,
+    endDelta,
+    view: { calendar },
+  }) => {
+    const { activityId } = fcEvent.extendedProps;
+
+    const startScaleDuration = toLuxonDuration(startDelta, calendar);
+    const startScaleIso = startScaleDuration.toISO();
+
+    const endScaleDuration = toLuxonDuration(endDelta, calendar);
+    const endScaleIso = endScaleDuration.toISO();
+
+    dispatch(scaleActivity(activityId, startScaleIso, endScaleIso));
+  };
+
+  const addActivityFromCalendar = (startLuxon, endLuxon) => {
+    setModalLuxonStart(startLuxon);
+    setModalLuxonEnd(endLuxon);
+
+    setShowActivityModal(true);
+  };
+
+  const addActivityFromCalendarClick = ({ date, view: { calendar } }) => {
+    const eventStartLuxon = toLuxonDateTime(date, calendar);
+    const eventEndLuxon = eventStartLuxon.plus({ minutes: defaultDurationFromActivityCode('other') });
+
+    addActivityFromCalendar(eventStartLuxon, eventEndLuxon);
+  };
+
+  const addActivityFromCalendarDrag = ({ start, end, view: { calendar } }) => {
+    const eventStartLuxon = toLuxonDateTime(start, calendar);
+    const eventEndLuxon = toLuxonDateTime(end, calendar);
+
+    addActivityFromCalendar(eventStartLuxon, eventEndLuxon);
+  };
+
+  const editCustomEvent = ({ event: fcEvent, view: { calendar } }) => {
+    const {
+      activity,
+      startLuxon,
+      endLuxon,
+    } = fcEventToActivityAndDates(fcEvent, calendar);
+
+    const canEdit = activity.activityCode.startsWith('other-');
+
+    if (canEdit) {
+      setModalActivity(activity);
+
+      setModalLuxonStart(startLuxon);
+      setModalLuxonEnd(endLuxon);
+
+      setShowActivityModal(true);
+    }
+  };
+
+  const onActivityModalClose = (ok, modalData) => {
+    setShowActivityModal(false);
+
+    const { activityCode, activityName } = modalData;
+
+    if (ok) {
+      if (modalActivity) {
+        dispatch(editActivity(modalActivity.id, 'activityCode', activityCode));
+        dispatch(editActivity(modalActivity.id, 'name', activityName));
+      } else {
+        const utcStartIso = luxonToWcifIso(modalLuxonStart);
+        const utcEndIso = luxonToWcifIso(modalLuxonEnd);
+
+        const activity = {
+          name: activityName,
+          activityCode,
+          startTime: utcStartIso,
+          endTime: utcEndIso,
+          childActivities: [],
+        };
+
+        dispatch(addActivity(activity, wcifRoom.id));
+      }
+    }
+
+    // cleanup.
+    setModalActivity(null);
+
+    setModalLuxonStart(null);
+    setModalLuxonEnd(null);
+  };
+
+  return (
+    <>
+      <Container textAlign="center">
+        <Dropdown fluid placeholder="Venue" clearable selection options={venueOptions} value={selectedRoomId} onChange={setSelectedRoomId} />
+      </Container>
+      {!!selectedRoomId && (
```

Comment:
I don't think the `!!` is needed; if it's falsey then it won't render, and if it's truthy then the fragment will display.